### PR TITLE
#54: Catch and log Slick exceptions

### PR DESCRIPTION
**Pull request by jozefbakus**
Implement #54 - Catch and log Slick exceptions

### CODE REVIEW

File: src/main/scala/za/co/absa/hyperdrive/trigger/persistance/EventRepository.scala [hunk 1]
```diff
@@ -32,15 +32,15 @@ class EventRepositoryImpl @Inject()(val dbProvider: DatabaseProvider) extends Ev
   import api._
 
   override def getAllEvents()(implicit ec: ExecutionContext): Future[Seq[Event]] = db.run(
-    eventTable.result
+    eventTable.result.withErrorHandling()
   )
 
   override def getExistEvents(sensorEventIds: Seq[String])(implicit ec: ExecutionContext): Future[Seq[String]] = db.run(
-    eventTable.filter(e =>  e.sensorEventId.inSet(sensorEventIds)).map(_.sensorEventId).result
+    eventTable.filter(e =>  e.sensorEventId.inSet(sensorEventIds)).map(_.sensorEventId).result.withErrorHandling()
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/hyperdrive/trigger/persistance/EventRepository.scala:39]: Sometimes you broke lines, and sometimes you didn't, like here. Would it make sense to properly format this by hand, or is it a useless exercise anyway without a code formatter?

**Review by kevinwallimann — COMMENTED** [review]
`.withErrorHandling` missing in methods:
- `switchWorkflowActiveState`
- `getWorkflowVersion`

File: src/main/scala/za/co/absa/hyperdrive/trigger/persistance/EventRepository.scala (hunk 1 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/persistance/EventRepository.scala:39]: For example this line is still reasonably long and fits into 120 chars. That was my reasoning when to broke lines.
Let's take scala style issue into next sprint.

**Comment by jozefbakus** [comment]
> `.withErrorHandling` missing in methods:
> 
> * `switchWorkflowActiveState`
> * `getWorkflowVersion`

Fixed

File: src/main/scala/za/co/absa/hyperdrive/trigger/persistance/EventRepository.scala (hunk 1 above)
Comment (kevinwallimann) [reply src/main/scala/za/co/absa/hyperdrive/trigger/persistance/EventRepository.scala:39]: Ah ok, it's about line length. Makes sense

**Review by kevinwallimann — DISMISSED** [review]

**Review by kevinwallimann — APPROVED** [review]

**Pull request merged**